File: python/potega.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-
#
#  potega.py


def potega(podst, wykladnik):
    wynik = 1
    for i in range(0, wykladnik):
        wynik = wynik * podst
    return wynik

# potega_rek(a, 0) = 1 dla a roznego od 0
# potega_rek(a, n) = potega_rek(a, n-1) * a dla n=N+


def potega_rek(a, n):
    if n == 0:
        return 1
    return potega_rek(a, n-1) * a


def main(args):
    a = float(input('Podaj podstawę: '))
    n = int(input('Podaj wykładnik: '))
    assert potega(1, 1) == 1
    assert potega(2, 1) == 2
    assert potega(3, 3) == 27
    assert potega(2, 2) == 4
    assert potega(5, 2) == 25

    print('Potęga: ', potega_rek(a, n))

    return 0

if __name__ == '__main__':
    import sys
    sys.exit(main(sys.argv))

File: cpp/potega.cpp
/*
 * potega.cpp
 * a0 = 1
 * a1 = a
 * an = a*...*a(n-czynnikow) dla n zawiera N+ -{1}
 */

#include <iostream>
 
using namespace std;
 
int potega_rek(int a, int n)
{
    if (n==0)
        return 1;
    return potega_rek(a, n-1) * a;
}


float potega(float a, int n)
{
    float wynik = 1;
 
    for (int i = 0; i<n; i++)
        wynik*=a;
 
    return wynik;
}
 
int main()
{
    int a;
    int n;
 
    cout << "Podaj podstawę: ";
    cin >> a;
    cout << "Podaj wykladnik: ";
    cin >> n;
 
    cout <<"Potęga: " << potega_rek(a, n)<<endl;
 
    return 0;
}
